src/cartesian/XAxis.tsx

```diff
@@ -49,4 +49,5 @@ XAxis.defaultProps = {
   scale: 'auto',
   reversed: false,
   allowDuplicatedCategory: true,
+  allowSelectNearestValue: false,
```

Comment:
nit: I know we already have `allowDuplicatedCategory` and `allowDataOverflow`, but this might be better as just `selectNearestValue`